File: Cpp/283_Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int index = 0;
        for(int i=0; i<nums.size(); ++i){
            if(nums[index] != 0){
                ++index;
            } else if(nums[i] != 0) {
                nums[index] = nums[i];
                nums[i] = 0;
                ++index;
            }
        }
    }
};

File: Kotlin/283_Move_Zeroes.kt
class Solution {
    fun moveZeroes(nums: IntArray): Unit {
        var index = 0
        for(i in nums.indices) {
            if(nums[index] != 0){
                ++index
            } else if(nums[i] != 0) {
                nums[index++] = nums[i]
                nums[i] = 0
            }
        }
    }
}